src/index.tsx

```diff
@@ -44,6 +38,23 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
                 onAfterPrint();
             }
         }, 500);
+    }
+
+    triggerPrint = (target) => {
+        const { onBeforePrint, onAfterPrint } = this.props;
+
+        if (onBeforePrint) {
+            const onBeforePrintOutput = onBeforePrint();
+            if (onBeforePrintOutput instanceof Promise) {
+                onBeforePrintOutput.then(r => {
```

Comment:
Nit: `r` isn't used and so doesn't need to be defined, should just be `()`

Reply:
Done.